.dotnet/src/Generated/Models/ToolChatMessage.cs

```diff
@@ -4,18 +4,22 @@
 
 using System;
 using System.Collections.Generic;
+using OpenAI;
 
 namespace OpenAI.Chat
 {
     public partial class ToolChatMessage : ChatMessage
     {
-        internal ToolChatMessage(ChatMessageRole role, ChatMessageContent content, IDictionary<string, BinaryData> serializedAdditionalRawData, string toolCallId) : base(role, content, serializedAdditionalRawData)
+        public ToolChatMessage(string toolCallId) : base(Chat.ChatMessageRole.Tool)
```

Comment:
Why is this public ctor generated here?

Reply:
This is an input model so we generate the ctor as public by default. It looks like the discrepancy here is that the base type is customized to be internal, so in Autorest we propagate that to the derived type. Need to consider whether we should adopt this for MGC or just add a customization for OpenAI.

Reply:
Fixed by customization